### DISCUSSION TITLE
Linking Connection Configurations to Exams

### DISCUSSION THREAD

**Discussion by michaelhq** [SafeExamBrowser/seb-server#90] [Q&A]
Hi all, I am currently working on a project at FHNW/BFH (with @lucaboesch) to implement a university-wide deployment of the SEB ecosystem. We are currently using SEB Server 1.5.1, SEB Browser Windows 3.5 & MacOS, iOS 3.3, Moodle 4.1.5. Everything is working fine and as expected.

Currently, I have a question regarding the use of Connection Configurations: Would it be possible to link on the SEB server defined exams with a specific connection configuration? Without exporting and distributing separate connection configurations per exam? Unfortunately, I have not found anything on how to do this.

Because if you have many exams running at the same time, this can quickly become confusing when starting the SEB and all exams were shown which were set up on the SEB Server in the exam selection dialog.

Regards,
Michael

**Comment by anhefti** [comment]
Hi Michael

Sure, this should be possible if you export an active Connection Configuration directly from the Connection Configuration page:
![image](https://github.com/SafeExamBrowser/seb-server/assets/39078777/04bbc795-47d9-419c-a372-0f2ae8779447)

If SEB loads such a SEB Server Connection Configuration, it connects to SEB Server as usual but it do no present a specified Exam for selection. In this case SEB Server sends a list of all currently running Exams (on SEB Server) to the SEB which presents it to the student/user for selection. After one has selected the exam within SEB, SEB loads the specific Exam Configuration from SEB Server and starts the exam as usual.

This are the two different main use-case for Connection Configuration with SEB Server.
Hope this helps, otherwise don't hesitate to ask further.

Regards
Andreas

**Comment by michaelhq** [comment]
Hello Andreas, thank you for your fast response! 

Yes, I know this function, how to export an active Connection Configuration directly. 

My question was not clear enough: Would it be possible to limit the number of exams that are displayed for a connection configuration in the SEB browser? So quasi a relationship between connection configuration and the exam, that only certain exams are displayed in the SEB.

The problem is that if you have a lot of running exams, you will see many entries in the SEB dialog. In a large organization with thousands of students, there may be 100 exams running at the same time.

This way you could create connection configurations for specific courses and classes, and only the current exams for these courses and classes would be displayed.

Best regards,
Michael

**Comment by anhefti** [comment]
Hi Michael

Aha, sorry about this misunderstanding. Unfortunately no, this is currently not possible and was also never requested yet. But I can see the utility of such a use-case. Technically it would be possible when one can (multi-select) exams for a Connection Configuration in the Connection Configuration page, then SEB Server only sends the selected exams for user-selection to SEB for a certain Connection Configuration.

I will create a feature request for this use-case and we will discuss this internally for prioritization and development planing. But earliest availability for such a  feature would be the next SEB Server version 1.6 that is planed for about Q2 24.

Best regards
Andreas
